bql/planner/data_access.go

```diff
@@ -595,30 +595,40 @@ func tripleToRow(t *triple.Triple, cls *semantic.GraphClause) (table.Row, error)
 		}
 	}
 	if cls.PAnchorBinding != "" {
+		var c *table.Cell
 		if p.Type() != predicate.Temporal {
-			// in the case of time anchor binding (eg: "bought"@[?time]) for an immutable predicate we just want to skip this triple and proceed to the next one, not returning any errors.
-			return nil, nil
-		}
-		t, err := p.TimeAnchor()
-		if err != nil {
-			return nil, fmt.Errorf("failed to retrieve the time anchor value for predicate %q in binding %q with error %v", p, cls.PAnchorBinding, err)
+			// In the case of time anchor bindings (eg: "bought"@[?time]) for immutable predicates we skip the triple if the clause is not optional, otherwise we provide an empty Cell as we want <NULL> to appear in the query result.
+			if !cls.Optional {
+				return nil, &skippableError{"cls.PAnchorBinding in non-optional clause", fmt.Errorf("tried to extract a time anchor binding from an immutable predicate")}
+			}
+			c = &table.Cell{}
+		} else {
+			t, err := p.TimeAnchor()
+			if err != nil {
+				return nil, fmt.Errorf("failed to retrieve the time anchor value for predicate %q in binding %q with error: %v", p, cls.PAnchorBinding, err)
+			}
+			c = &table.Cell{T: t}
 		}
```

Comment:
suggestion: This part of the code is already very indented. We can avoid one indentation if we avoid the else and always initialize the cell. Like:

c: = &table.Cell{}
if type != Temporal && !cls.Optional{
 return 'non-optional clause error'
}
if type == Temporal{
// set the 'T' on the cell
}
// We handle the table cell

This way we make clear our intend of when is the error path and when we fill the 'cell' struct.

Reply:
We discussed and decided, for now, to let it as it is as it makes more immediate for who is reading to understand the logic of the behavior in each case detailed in the PR description (for `cls.OAnchorBinding` for example the logic is: 1. if the object is non-predicate; 2. if it is an immutable predicate; 3. if it is a temporal predicate (as expected); adjusting the behavior when appropriate taking into account if the clause is optional or not).

But these hints commented here will also be taken into account for future situations to simplify and make complex `if else` statements more compact.